envi/archs/arm/disasm.py

```diff
@@ -1726,13 +1785,17 @@ def render(self, mcanv, op, idx):
         mcanv.addNameText(arm_regs[self.shreg][0], typename='registers')
 
     def repr(self, op):
-        rname = arm_regs[self.reg][0]+","
-        return " ".join([rname, shift_names[self.shtype], arm_regs[self.shreg][0]])
+        rname = arm_regs[self.reg][0]+", "
+        if self.display_shift:
+            rname+=shift_names[self.shtype]
+        rname+= arm_regs[self.shreg][0]
```

Comment:
if we end up keeping this, please make this a list that is then turned into only one string, as the previous version (almost) did.